drivers/core_demographics_report/app/models/core_demographics_report/document_exports/core_demographics_excel_export.rb

```diff
@@ -0,0 +1,79 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module CoreDemographicsReport::DocumentExports
+  class CoreDemographicsExcelExport < ::GrdaWarehouse::DocumentExport
+    include ApplicationHelper
+    def authorized?
+      user.can_view_any_reports? && report_class.viewable_by(user)
+    end
+
+    protected def report
+      @report ||= report_class.new(filter)
+    end
+
+    protected def view_assigns
+      comparison_filter = filter.to_comparison
+      comparison_report = report_class.new(comparison_filter) if report.include_comparison?
+
+      {
+        report: report,
+        filter: filter,
+        comparison: comparison_report || report,
+        comparison_filter: comparison_filter,
+        pdf: false,
+      }
+    end
+
+    def perform
+      with_status_progression do
+        ActionController::Renderer::RACK_KEY_TRANSLATION['warden'] ||= 'warden'
+        warden_proxy = Warden::Proxy.new({}, Warden::Manager.new({})).tap do |i|
+          i.set_user(user, scope: :user, store: false, run_callbacks: false)
+        end
+
+        renderer = controller_class.renderer.new(
+          'warden' => warden_proxy,
+        )
+
+        write_tmp_file(
+          renderer.render(
+            action: :index,
+            format: :xlsx,
+            assigns: view_assigns,
+          ),
+          "Core Demographics - #{Time.current.to_s(:db)}",
+        ) do |io|
+          self.pdf_file = io
+        end
+      end
+    end
+
+    def pdf_file=(file_io)
```

Comment:
Maybe not for today -- but a TODO -- we should rename this method if we are extending beyond PDF generation. Just `file`? Or `downloadable_file`?